MoreLinq/CountMethods.cs

```diff
@@ -0,0 +1,159 @@
+        }
+
+        /// <summary>
+        /// Determines whether or not the number of elements in the sequence is equals to the given integer.
+        /// </summary>
+        /// <typeparam name="T">Element type of sequence</typeparam>
+        /// <param name="source">The source sequence</param>
+        /// <param name="length">The exactly number of items a sequence must have for this
+        /// function to return true</param>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="length"/> is negative</exception>
+        /// <returns><c>true</c> if the number of elements in the sequence is equals
+        /// to the given integer or <c>false</c> otherwise.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = { 123, 456, 789 };
+        /// var result = numbers.Exactly(3);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>true</c>.
+        /// </example>
+        public static bool Exactly<T>(this IEnumerable<T> source, int length)
+        {
+            if (length < 0) throw new ArgumentOutOfRangeException("length", "length must not be negative.");
+
+            return QuantityIterator(source, length + 1, count => count == length);
+        }
+
+        /// <summary>
+        /// Determines whether or not the number of elements in the sequence is between 
+        /// an inclusive range of minimum and maximum integers.
+        /// </summary>
+        /// <typeparam name="T">Element type of sequence</typeparam>
+        /// <param name="source">The source sequence</param>
+        /// <param name="min">The minimum number of items a sequence must have for this
+        /// function to return true</param>
+        /// <param name="max">The maximun number of items a sequence must have for this
+        /// function to return true</param>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="min"/> is negative or <paramref name="max"/> is less than min</exception>
+        /// <returns><c>true</c> if the number of elements in the sequence is between (inclusive)
+        /// the min and max given integers or <c>false</c> otherwise.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = { 123, 456, 789 };
+        /// var result = numbers.CountBetween(1, 2);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>false</c>.
+        /// </example>
+        public static bool CountBetween<T>(this IEnumerable<T> source, int min, int max)
+        {
+            if (min < 0) throw new ArgumentOutOfRangeException("min", "min must not be negative.");
+            if (max < min) throw new ArgumentOutOfRangeException("max", "max must be greater than or equals to min.");
```

Comment:
Replace:

    max must be greater than or equals to min.

with:

    Maximum count must be greater than or equal to the minimum count.